### PR TITLE
Added Merge Sort in C#

### LINKED ISSUE #1630: Merge Sort implementation in C#
I would like to add Merge Sort in C# under GSSoc'20

### PR DESCRIPTION
**Pull request by Yashank18**
Fixes #1630 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Implementation of Merge Sort

#### Languages Used:
- C#

#### Files Added:
- Merge_sort.cs

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by Yashank18** [comment]
Okay, I will work on it and get back to you.

**Comment by Yashank18** [comment]
@singh-shreya6 Updated my PR as requested :)

File: Merge_Sort/Merge_sort.cs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    }
+
+    public static void Main()
+    {
+    	int size;
+    	size= Convert.ToInt32(Console.ReadLine()); 
+
+        int[] array = new int[size];
+        for(int i=0; i<size; i++)  
+    {
```
Comment (singh-shreya6) [inline Merge_Sort/Merge_sort.cs]: Please fix indention.

File: Merge_Sort/Merge_sort.cs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        for(int i = 0; i < size; i++)
+            Console.Write(array[i] + " ");
+
+        Console.Write("");
+    }
+
+    public static void Main()
+    {
+    	int size;
```
Comment (singh-shreya6) [inline Merge_Sort/Merge_sort.cs]: Add 4 space indentation

File: Merge_Sort/Merge_sort.cs (hunk 1 above)
Comment (Yashank18) [reply Merge_Sort/Merge_sort.cs]: Okay, working on it

**Comment by Yashank18** [comment]
@singh-shreya6  Updated my PR. Please review

File: Merge_Sort/Merge_sort.cs [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+It divides input array in two halves, calls itself for the two halves and then merges the two sorted halves. 
+The conquer_merge() function is used for merging two halves.
+*/
+
+using System;
+public class Merge_Sort
+{
+    public static int[] temp;
+
+	 // Conquer
```
Comment (singh-shreya6) [inline Merge_Sort/Merge_sort.cs]: Indent this comment properly.

**Comment by Yashank18** [comment]
Okay, I will do that

**Comment by Yashank18** [comment]
@singh-shreya6  Please Check

**Comment by vaishnavi-janardhan** [comment]
@singh-shreya6 Can you please add labels `gssoc20` and one of `beginner`, `easy`, `medium` or `hard` to this PR.

**Comment by Yashank18** [comment]
@singh-shreya6  Please add Labels

**Pull request merged**